test/functional/connection.test.js

```diff
@@ -273,4 +274,32 @@ describe('Connection', function () {
       done();
     }
   });
+
+  it(
+    'should be able to connect again after close',
+    withClient(function (client, done) {
+      expect(client.isConnected()).to.be.true;
+
+      const collection = () => client.db('testReconnect').collection('test');
+      collection().insertOne({ a: 1 }, (err, result) => {
+        expect(err).to.not.exist;
+        expect(result).to.exist;
+
+        client.close(err => {
+          expect(err).to.not.exist;
+
+          client.connect(err => {
+            expect(err).to.not.exist;
```

Comment:
And another `expect(client.isConnected()).to.be.true;` here, just as a sanity check.